src/managers/KeybindManager.cpp

```diff
@@ -1153,14 +1181,38 @@ void CKeybindManager::moveFocusTo(std::string args) {
     const auto PWINDOWTOCHANGETO = PLASTWINDOW->m_bIsFullscreen ? g_pCompositor->getNextWindowOnWorkspace(PLASTWINDOW, arg == 'u' || arg == 't' || arg == 'r') :
                                                                   g_pCompositor->getWindowInDirection(PLASTWINDOW, arg);
 
+    // Found window in direction, switch to it
     if (PWINDOWTOCHANGETO) {
         switchToWindow(PWINDOWTOCHANGETO);
-    } else {
-        const auto PWINDOWNEXT = g_pCompositor->getNextWindowOnWorkspace(PLASTWINDOW, true);
-        if (PWINDOWNEXT) {
-            switchToWindow(PWINDOWNEXT);
+        return;
+    }
+
+    Debug::log(LOG, "No window found in direction %c, looking for a monitor", arg);
+
+    // No window found in direction, switch to workspace in direction
+    const auto PNEWMONITOR = g_pCompositor->getMonitorInDirection(arg);
+    if (PNEWMONITOR && PNEWMONITOR->activeWorkspace != -1) {
+        const auto PNEWWORKSPACE = g_pCompositor->getWorkspaceByID(PNEWMONITOR->activeWorkspace);
+        if (PNEWWORKSPACE) {
+            g_pCompositor->focusWindow(nullptr);
+            g_pCompositor->setActiveMonitor(PNEWMONITOR);
+            g_pCompositor->deactivateAllWLRWorkspaces(PNEWWORKSPACE->m_pWlrHandle);
+            PNEWWORKSPACE->setActive(true);
+
+            Vector2D middle = PNEWMONITOR->vecPosition + PNEWMONITOR->vecSize / 2.f;
+            g_pCompositor->warpCursorTo(middle);
+
+            return;
         }
     }
+
+    Debug::log(LOG, "No monitor found in direction %c", arg);
+
+    // TODO: Add option to disable looping/falling back to any next window?
```

Comment:
I'd rather not yeet it atm if we don't have the option

Reply:
Can I just add an option? Cause I personally don't want that at all, so while I'm at it

Reply:
sure